DigitalLearningSolutions.Data/Helpers/ExternalApis/LearningHubApiClient.cs

```diff
@@ -0,0 +1,153 @@
+﻿namespace DigitalLearningSolutions.Data.Helpers.ExternalApis
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using DigitalLearningSolutions.Data.Models.LearningHubApiClient;
+    using Microsoft.Extensions.Configuration;
+    using Newtonsoft.Json;
+
+    public interface ILearningHubApiClient
+    {
+        Task<ResourceSearchResult> SearchResource(
+            string text,
+            int? offset = null,
+            int? limit = null,
+            IEnumerable<string>? resourceTypes = null
+        );
+
+        Task<ResourceReferenceWithReferenceDetails> GetResourceByReferenceId(int resourceReferenceId);
+
+        Task<BulkResourceReferences> GetBulkResourcesByReferenceIds(
+            IEnumerable<int>? resourceReferenceIds = null
+        );
+    }
+
+    public class LearningHubApiClient : ILearningHubApiClient
+    {
+        private readonly HttpClient client;
+        private readonly string learningHubApiBaseUrl;
+
+        public LearningHubApiClient(HttpClient httpClient, IConfiguration config)
+        {
+            string learningHubApiKey = config.GetLearningHubApiKey();
+            learningHubApiBaseUrl =
+                "https://uks-learninghubnhsuk-openapi-dev.azurewebsites.net";

```

Comment:
Added the url to the appsettings files, is that the right place for it? It's not a user secret is it?

Reply:
I would assume this is fine, might be best to just check with Steve before we merge @SteveJacksonSoft

Reply:
This should go in config.
Generally it's best to have URLs of external dependencies in config because they might change, and also in this case different envs will point to different envs of Learning Hub Open API.
Can we include LearningHub**Open**Api in the name of the config value? LH have two APIs, and anyone on the LH team will think of the other one when you say "LearningHubApi"?

Reply:
But it's not secret. Thanks for asking though.